### ISSUE TITLE
Support having .git/hooks as a volume

### ISSUE THREAD

**Issue by botanicus** [botanicus/docker-project-manager#18]
After that, recreate the env for DPM and add `pre-commit` hook to run `crystal tool format` in it.

**Comment by botanicus** [comment]
Another issue is how do we do run the initial `git clone`? (The host system doesn't have `git`.)

- **New project:** straighforward: run the `dev` shell, and in there, the development image provides `git`. So all there is to do is to run `git init`.
- **Existing project:** if we want to clone an existing project, what I was doing was:

```
dpm attach
git clone repo X
mv X.git my_repo/
cd my_repo
git checkout
```

Not the best. I suspect it might be shortened to:

```
dpm attach
git clone repo my_repo/.git --bare
cd my_repo
git checkout
```

Better, but far from perfect. The main thing is _only after we clone the repo, we can add `my_repo/.git/hooks` as a volume_.

At the same time, we cannot quite do it in the same time as the init command runs: at least not without making it interactive, as we have to add the SSH keys to GH/GL before we can clone a private repo.

There are some `libgit2` bindings for Crystal, it'd be possible to introduce `dpm clone <repo-url>`. That would also extend `Dockerfile` with `VOLUME my-repo/.git/hooks`.

For now I think the best is to just print the instructions.
